bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/handler/ICloudDeviceHandler.java

```diff
@@ -147,14 +154,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void dispose() {
-        ICloudAccountBridgeHandler bridge = getIcloudAccount();
+
+        Bridge bridge = getBridge();
         if (bridge != null) {
-            bridge.unregisterListener(this);
+            ((ICloudAccountBridgeHandler) bridge.getHandler()).unregisterListener(this);
```

Comment:
You could probably even do this:
```java
ThingHandler bridgeHandler = bridge.getHandler();
if (bridgeHandler instanceof ICloudAccountBridgeHandler)
{
    ((ICloudAccountBridgeHandler) bridgeHandler).unregisterListener(this);
}
```

to have both null and type check at the same time.